src/PreferencesDialog.cpp

```diff
@@ -434,6 +455,19 @@ void PreferencesDialog::removeExtension()
         ui->listExtensions->takeItem(ui->listExtensions->currentIndex().row());
 }
 
+void PreferencesDialog::listUpBuiltinExtension()
+{
+    const QDir dir(qApp->applicationDirPath() + "/../Extensions/");
+    QStringList files = dir.entryList(QStringList() << "*.dylib", QDir::Files);
```

Comment:
I think this is the only part preventing this new feature to be portable. The "*.dylib" could be made OS-dependent and all the `#ifdef Q_OS_MACX` removed.

Reply:
Aren't we starting with doing these on macOS first, and then expanding out to other OS's if it goes well?

Reply:
Yes, I was actually going to implement the feature for Windows after I finished this PR, but i's okay to do it in this PR. However, I'm not sure if Linux has a place in the executable file to store the extension library.

Reply:
At least on Debian/Ubuntu, if you install an SQLite extension, it will be placed under `/usr/lib/x86_64-linux-gnu/mod_*.so`. Thus, it would make sense to use the path `qApp->applicationDirPath() + "/../" + QSysInfo::currentCpuArchitecture + "-linux-gnu/mod-*.so". With that, when the user installs the official Debian sqlitebrowser package and any SQLite extension package, the extension will be listed as an available extension to load. In that way, we will save the user from having to search the extension after the installation.

If this defers per Linux distribution, we might choose to support only one, or try to detect the distribution using [`QSysInfo::productType()`](https://doc.qt.io/qt-6/qsysinfo.html#productType).

No problem if this is done for macOS only because it is urgent for the release, and then generalized for other OS's. But in general, we should prefer to make everything as common and portable as possible.

Reply:
Thank you so much. This will go a long way toward to resolving the issue.
Let's take a look at a few popular Linux distributions and see where other distributions store their extensions.

As an aside, I personally think it's better not to include this PR in this release. 🤔

Reply:
I've changed the code to support cross-platform, but there's something to think about. On Linux, the list of extensions output by the program is too long because we refer to the general paths used by the system. Should we change it to refer to a path that contains only the extensions we use?

Perhaps we could take a cue from @justinclift and have the package installation process install our extension to a specific path (e.g. `/usr/lib/sqlite-extension`).

![image](https://github.com/sqlitebrowser/sqlitebrowser/assets/64626662/42188293-c954-4495-8d95-17051f48c715)

Reply:
The list you're showing in that dialog is pretty much a dump of all entries in the shared library directory.  That's not going to be correct. :wink:

> Should we change it to refer to a path that contains only the extensions we use?
> Perhaps we could ... have the package installation process install our extension to a specific path

Yeah.  We should probably adjust our packaging on Linux to include the extensions in some subdirectory we create.  No idea where exactly (yet), but it shouldn't be too hard to figure out a reasonable spot.

With that done then we can list the extensions present in that directory.

* There are two packages we create ourselves, the AppImage and the Snap (@deepsidhu1313)
* For distribution created packages (Debian, etc), we could ping the maintainers and ask them for their thoughts.

Reply:
Yes, I don't know much about Linux-side distributions, so I'll wait for others to chime in. 😄

Reply:
You're missing the prefix used by SQLite extensions (`mod_*.so`). On my system:
```sh
$ ls /usr/lib/x86_64-linux-gnu/mod_*.so
/usr/lib/x86_64-linux-gnu/mod_spatialite.so
$ ls /usr/lib/x86_64-linux-gnu/lib*.so | wc -l
261
```

Reply:
Oops, just saw that libsqlite3-mod-csvtable uses a different convention: `/usr/lib/x86_64-linux-gnu/libsqlite3_mod_csvtable.so`. So not so easy, but maybe just a matter of using `*mod_*.so`, but might be fragile.

Reply:
Checking list of packages on Debian, most of them are libsqlite3_mod_*.so, but there are a few weird cases, like https://packages.debian.org/sid/amd64/libsqlite3-mod-ceph/filelist

Reply:
I didn't realize I needed to add that prefix. Thanks for letting me know.

But then I thoguht, this is a PR that implements auto-loading for the built-extensions that our application provides, so maybe we don't need to automatially load SQLite extensions installed by external packages on the user's system? (If user need it, user can still add that extension in the Preferences today)

Alternatively, I think we could implement an additional feature to find and automatically load SQLite extensions that users have installed on their system. (Maybe function name will be 'Load System-wide Installed SQLite Extensions). 🤔

Reply:
@mgrojo I'd like to wrap up this PR, but in case you missed the notification, so I'm pinging you again. :)

Reply:
Sorry for the delay, @lucydodo. I've taken a new look and tested it on my Linux.

The Linux implementation was including all the system shared libraries in the list, turning it inconvenient for the user. I've made a change that filters the list using convention file patterns for SQLite extensions. I've also added support for Fedora, which uses a different location for libraries. It's a pity that there is not a unified naming convention for SQLite extensions, or a standard location for them, but at least the feature will be useful for Linux users, who will have an easy way to install the SQLite extensions after installing them using the OS package manager.

I've tested this only on my distribution, though, which is Ubuntu, but the difference is trivial, nevertheless.

Reply:
Cool, sounds like a good improvement @mgrojo. :smile:

Reply:
This is how it looks on my system after installing all the packages of SQLite extensions I found on Ubuntu:

![imagen](https://github.com/sqlitebrowser/sqlitebrowser/assets/13812910/f1b3936a-faf4-4462-824d-e30a5634ed8e)

At first, I wanted to exclude the strangely versioned files used for some of the extensions, but the glob file pattern matching was not working. In any case, users might want to load specific versions in case they have more than one. In my case there was only one version available, being the unversioned file just a link to the former.

I approve this PR, but it seems I cannot tell it to GitHub, maybe because there are conflicts to solve.

Reply:
That's a cool approach. Thank you. I just resolved the conflict and merged it. :)